Adding a fix for lcov/gcc8

milljm opened the pull request:
This fixes bug: https://github.com/linux-test-project/lcov/issues/38

Commit:
Adding a fix for lcov/gcc8

This fixes bug: https://github.com/linux-test-project/lcov/issues/38

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/325748/) on d87780c wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/198/site)

This comment will be updated on new commits.

Commit:
Fix lcov, clang's ld, and urls.txt

Latest Lcov includes a makefile now. Use that.

Append path to llvm's bin directory instead of pre-prending when loading the the GCC module. This is necessary due in part to the addition of clang's linker we now build.

Fix that pesky issue when urls.txt is missing, conda create environment fails, unless you own the conda installation.

Commit:
Modules does not support altering existing paths

Once you set a path using modules, say with append-path. But then another
module says to prepend-path that same path, Modules does not alter said
path.

So, our GCC module, while needing to be loaded when Clang is loaded,
conflicts with our path requirements.

Thankfully we no longer need the clang-formatter. Remove append-path.

Commit:
Fix Darwin Xcode 10.2 __Atomic bug

Fix clobbering issue of a users bash profile for OS X.

Commit:
I knew it was not applying...

Remove the no longer necessary jit patch.

Commit:
Upgrade to GCC 8.3.0

The __Atomic patch supplied by homebrew is for GCC 8.3.0. Let us try to
build that version instead.

milljm commented:
Merging even though this is reporting failure. This is due to the moose group needs a _failing_ pr into moose, _with_ this error, in order to troubleshoot it.
